### ISSUE TITLE
[Bug]: ConcurrentModificationException - com.onesignal.common.modeling.Model.initializeFromModel(Model.kt:123)

### ISSUE THREAD

**Issue by tajchert** [OneSignal/OneSignal-Android-SDK#1819]
### What happened?

Crash at user device. 24 users affected, all Android versions

### Steps to reproduce?

```Markdown
-
```

### What did you expect to happen?

No crash.

### OneSignal Android SDK version

5.0.0

### Android version

13, 12, 11, 10, 9

### Relevant log output

```Shell
Fatal Exception: java.util.ConcurrentModificationException:
       at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:757)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:790)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:788)
       at com.onesignal.common.modeling.Model.initializeFromModel(Model.kt:123)
       at com.onesignal.core.internal.config.impl.ConfigModelStoreListener$fetchParams$1.invokeSuspend(ConfigModelStoreListener.kt:69)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:33)
       at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(Builders.kt:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(Builders.kt:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tajchert** [comment]
This seems similar to https://github.com/OneSignal/OneSignal-Android-SDK/issues/1789 but stacktrace is different.

**Comment by jennantilla** [comment]
@tajchert thank you for letting us know--we're looking into this and will get back to you as soon as possible.

**Comment by dginovker** [comment]
We're seeing the same thing on our Android Unity games

**Comment by dmytroreutov** [comment]
I'm facing same crash for `com.onesignal:OneSignal:[5.0.0, 5.99.99]`

```
Fatal Exception: java.util.ConcurrentModificationException:
       at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:760)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:793)
       at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:791)
       at com.onesignal.common.modeling.Model.initializeFromModel(Model.kt:123)
       at com.onesignal.core.internal.config.impl.ConfigModelStoreListener$fetchParams$1.invokeSuspend(ConfigModelStoreListener.kt:69)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.internal.ScopeCoroutine.afterResume(Scopes.kt:33)
       at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(BlockingCoroutine.java:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(BuildersKt__Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(Builders.kt:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(BuildersKt__Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(Builders.kt:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

**Comment by nan-li** [comment]
Hi all, I apologize for any delay,

We have identified the cause and are working on the fix for all the methods this crash is reported.

**Comment by jennantilla** [comment]
Hello everyone! Thank you for your patience here. We have introduced a fix for this crash in our latest release, [5.0.4](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.0.4). Please update your SDK and let us know if you are still having any issues.

**Comment by jennantilla** [comment]
Closing out this issue. Please let us know if you are seeing any further occurrences of this exception. Thanks!
